apps/dashboard/src/app/login/LoginPage.tsx

```diff
@@ -154,6 +154,32 @@ function PageContent(props: {
     router.replace(props.redirectPath);
   }
 
+  async function onLogin() {
+    const account = await getRawAccountAction();
+
+    // shouldn't happen - but if account is not found, stay on login page
+    if (!account) {
+      return;
+    }
+
+    if (!isOnboardingComplete(account)) {
+      setScreen({
+        id: "onboarding",
+        account,
+      });
+    } else {
+      onComplete();
+    }
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  useEffect(() => {
+    // if suddenly disconnected
+    if (connectionStatus !== "connected" && screen.id !== "login") {
+      setScreen({ id: "login" });
```

Comment:
is this safe?

Reply:
Yeah - just making sure that you are never setting an onboarding screen if wallet is disconnected - login screen is the initial ConnectEmbed UI